### PR TITLE
Enocde emails

### PR DESCRIPTION
**Pull request by aandis**
Just like https://github.com/gratipay/gratipay.com/pull/3965 :)

### CODE REVIEW

**Comment by aandis** [comment]
Haven't tested this locally yet. I can't sign in. :/
Also many of `sign_in` tests are failing for me locally.

File: tests/py/test_email.py [hunk 1]
```diff
@@ -108,6 +116,17 @@ def test_verify_email(self):
         actual = Participant.from_username('alice').email
         assert expected == actual
 
+    def test_email_verification_is_backwards_compatible(self):
+        """Test email verification still works with unencoded email in verification link.
+        """
+        self.hit_email_spt('add-email', 'alice@example.com')
+        nonce = self.alice.get_email('alice@example.com').nonce
+        url = '/alice/emails/verify.html?email=alice@example.com&nonce='+nonce
+        self.client.GET(url, auth_as='alice')
```
Comment (Changaco) [inline tests/py/test_email.py]: Our `auth_as` is different from Gratipay's, it takes a `Participant` object, not a username.

Comment (aandis) [inline tests/py/test_email.py]: Ah sorry. Missed it between other failing tests. I have `6 failed, 377 passed in 88.10 seconds` event after fixing this. All from `tests/py/test_sign_in.py F...F..F.FFF..........`

Comment (aandis) [inline tests/py/test_email.py]: Fixed in 359ec6798661e764dde7aa28c2afbe961a1f412a

Comment (Changaco) [inline tests/py/test_email.py]: > I have `6 failed, 377 passed in 88.10 seconds` event after fixing this. All from `tests/py/test_sign_in.py F...F..F.FFF..........` Will investigate that later.

Okay, don't hesitate to post details in https://github.com/liberapay/salon/issues/26 or in a new issue if you need some help figuring out the problem. ;-)

**Comment by Changaco** [comment]
I've made cosmetic changes (reordered some imports alphabetically, squashed a commit).

**Pull request merged**